mm2src/coins/eth/nft/nft_structs.rs

```diff
@@ -0,0 +1,220 @@
+use crate::{TxFeeDetails, WithdrawFee};
+use mm2_number::BigDecimal;
+use rpc::v1::types::Bytes as BytesJson;
+use serde::Deserialize;
+use std::str::FromStr;
+
+#[derive(Debug, Deserialize)]
+pub struct NftListReq {
+    pub(crate) chains: Vec<Chain>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct NftMetadataReq {
+    pub(crate) token_address: String,
+    pub(crate) token_id: BigDecimal,
+    pub(crate) chain: Chain,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum Chain {
+    Bnb,
+    Eth,
+}
+
+#[derive(Debug, Display)]
+pub enum ParseContractTypeError {
+    UnsupportedContractType,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub enum ContractType {
+    Erc1155,
+    Erc721,
+}
```

Comment:
`ContractType` deserializes from `"Erc1155"`, but `ContractType::from_str` expects `ERC1155`.
How about specifying `#[serde(rename_all = "uppercase")]`.

Reply:
Ooops, missed it, thanks. Added "UPPERCASE" for `Chain` too. Think it looks better.
Done